dlt-connector/src/typeorm/DataSource.ts

```diff
@@ -4,23 +4,86 @@ import { DataSource as DBDataSource, FileLogger } from '@dbTools/typeorm'
 import { entities } from '@entity/index'
 
 import { CONFIG } from '@/config'
+import { logger } from '@/server/logger'
+import { Migration } from '@entity/Migration'
+import { LogError } from '@/server/LogError'
 
-const DataSource = new DBDataSource({
-  type: 'mysql',
-  host: CONFIG.DB_HOST,
-  port: CONFIG.DB_PORT,
-  username: CONFIG.DB_USER,
-  password: CONFIG.DB_PASSWORD,
-  database: CONFIG.DB_DATABASE,
-  entities,
-  synchronize: false,
-  logging: true,
-  logger: new FileLogger('all', {
-    logPath: CONFIG.TYPEORM_LOGGING_RELATIVE_PATH,
-  }),
-  extra: {
-    charset: 'utf8mb4_unicode_ci',
-  },
-})
-
-export const getDataSource = () => DataSource
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class Connection {
+  // eslint-disable-next-line no-use-before-define
+  private static instance: Connection
+  private connection: DBDataSource
```

Comment:
I use the Singleton Pattern here: https://refactoring.guru/design-patterns/singleton
Chat GPT Comment on you suggestion: 
"Introducing a factory class and splitting responsibilities might seem like an over-engineering if your singleton class is straightforward and doesn't require significant changes or additional functionality related to instantiation."

I don't see an advantage to create a Factory Class for creating the db connection. More a disadvantage. Because I must make sure that the function is really called once. I don't know how typeorm will react if more than one connection was created. 

I have build this class after different tries to have a easy solution for mocking db in the tests. This solution work best so far.

Reply:
Sorry, i think now i recognize my difficulties about your solution...

The class name is `Connection`
the instance is of type `Connection`
the DBDataSource-attribute is named `connection`

and i thought you have a static connection for the singleton instance and inside this instance again a connection is instance-attribut.

Your solution is ok as it is. Sorry...